tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+            [-1.00631180e-06,  9.85991203e-05,  4.33475295e-05, -6.09100454e-05, -6.60120574e-05],
+            [-2.17261186e-04, -4.51273621e-05, -1.30990539e-04,  6.68971521e-05, -4.87438362e-05],
+            [-1.09714163e-04, -2.69894566e-04,  2.18208856e-05, -6.44599929e-05, -1.96370002e-04],
+            [-7.71549133e-05, -7.37010252e-05, -4.22884340e-05, -2.25038388e-04, -1.66657239e-04],
+            [-3.36196989e-06, -1.18451671e-04, -4.52626978e-06,  1.02981866e-04, -8.37329754e-05],
+            [ 3.53793964e-05,  4.54964533e-05, -6.34722823e-05, -1.94625852e-05,  1.89677483e-05],
+            [ 1.66565383e-04,  3.19224191e-06, -2.31259304e-05,  1.85983791e-05,  3.81422875e-05],
+            [ 9.28366905e-05,  1.74423406e-04, -7.30022809e-07, -5.36855858e-05,  1.13245124e-04],
+            [ 0.00016175,      0.00019267,      0.00023797,      0.00010066,      0.00010137    ],
+            [ 1.13502277e-04,  4.48093004e-05, -1.08588317e-04,  7.08992725e-06,  9.79580181e-05],
+            [ 5.45503135e-05,  4.60378864e-05,  7.31752847e-05,  5.34881056e-05, -4.42780821e-05],
+            [-2.93195776e-04,  4.56038593e-05,  9.75722094e-05,  1.47418646e-04, -1.62329742e-04],
+            [ 4.76921830e-05,  4.01996255e-05, -3.23069382e-05,  1.35737054e-04,  2.14109414e-06],
+            [ 1.13051655e-04,  9.03226722e-05,  8.25028066e-05, -2.57863583e-06,  4.62620049e-05],
+            [ 1.18207098e-04,  1.39375924e-05,  1.87378588e-04, -3.51928983e-06, -4.70459273e-05],
+            [ 1.28281884e-04,  2.50304808e-05, -7.40293647e-05,  1.62979923e-04,  1.20208384e-04],
+            [ 3.80926384e-05,  6.51673450e-05,  5.67955276e-05, -4.40645390e-05,  5.41397444e-05],
+            [-1.65321007e-04, -2.43259429e-04,  1.10643723e-04,  6.81763658e-05,  8.39810614e-05],
+            [ 5.30247326e-05,  9.02680397e-05,  6.68931022e-05, -4.69935188e-05, -4.93068724e-05],
+            [ 9.43813269e-05, -1.30082385e-04,  1.47130724e-04, -6.13797435e-07, -4.42286161e-06],
+            [ 6.62178755e-05, -1.50319069e-05,  2.87163302e-05,  7.30216841e-05, -5.44442313e-05],
+            [-1.06548633e-04, -4.39539406e-05,  7.23699111e-05, -1.26276598e-04,  1.94361202e-04],
+            [-9.22824472e-05, -1.11743463e-04,  2.87895278e-05,  4.79988615e-05,  4.64047814e-05],
+            [ 1.75375175e-04, -4.40491161e-05, -6.82584569e-05, -1.13435450e-04, -2.26946285e-05],
+            [-7.39408960e-05, -2.65184279e-05,  5.78177851e-05,  1.17788935e-05, -2.97591011e-05],
+            [ 1.00971288e-04, -1.67156860e-04, -5.24639395e-05,  8.93173061e-05,  2.16047623e-05],
+            [ 2.06426457e-04,  8.27577854e-05, -4.61942718e-05, -6.92981952e-05,  4.04636422e-05]
+        ])
+
+        A = BlkDiagMatrix.from_blk_diag(sn_matrix)
+        coeff_est = A.solve(coeff)
+        self.allallfunc(result, coeff_est)
+
+    def testBlkDiagMatrixTranspose(self):
+        blk_c = self.blk_a.T
+        self.allallfunc(blk_c, self.blk_a)
```

Comment:
If it's transposed, it normally does not equal itself. I'm guessing these matrices are symmetric? That may cause tests to miss some bugs that only occur for non-symmetric matrices.

Reply:
They are. Sure I can improve the original tests. There are a couple other things I can do together.

Reply:
Ok, let's make that a separate issue along with the other test-related changes.